### PR TITLE
Adds hashing method function

### PR DESCRIPTION
**Pull request by luispresuelVenafi**
Adding SHA1SUM method function in utils for soon usage `store_by=hash`

### CODE REVIEW

File: plugin/pki/util.go [hunk 1]
```diff
@@ -600,3 +604,14 @@ func shortDurationString(d time.Duration) string {
 	}
 	return s
 }
+
+func sha1sum(s string) string {
+	hash := sha1.New()
+	buffer := []byte(s)
+	hash.Write(buffer)
+	return hex.EncodeToString(hash.Sum(nil))
```
Comment (rvelaVenafi) [inline plugin/pki/util.go:612]: Why passing nil as argument here? shouldnt the buffer be passed as argument?

File: plugin/pki/util_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+func TestSHA1SUM(t *testing.T) {
+	// Known SHA1SUM value of "hello"
+	// echo -n "hello" | sha1sum
+	SHA1SUMstringValue := "aaf4c61ddcc5e8a2dabede0f3b482cd9aea9434d"
+
+	s := "hello"
+	SHA1SUMvalue := sha1sum(s)
+	if SHA1SUMstringValue != SHA1SUMvalue {
+		t.Fatalf("%s function", GetFunctionName(sha1sum))
```
Comment (rvelaVenafi) [inline plugin/pki/util_test.go:15]: Why a function to extract the name of the sha1 function is rrequired here? Shouldnt a fixed value be better?
Even better shouldn't a meaningful error message be returned here?

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]

File: plugin/pki/util.go (hunk 1 above)
Comment (EduardoVV) [reply plugin/pki/util.go:612]: The buffer is already populated with the `hash.Write` function

File: plugin/pki/util_test.go (hunk 2 above)
Comment (luispresuelVenafi) [reply plugin/pki/util_test.go:15]: I was trying to denote which function is the one failing. But yes, I should have put a better error message

File: plugin/pki/util.go (hunk 1 above)
Comment (luispresuelVenafi) [reply plugin/pki/util.go:610]: @rvelaVenafi , @EduardoVV already answered what I was going to say

**Review by rvelaVenafi — APPROVED** [review]

**Review by EduardoVV — APPROVED** [review]

**Pull request merged**